src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/RuleExporter.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Rules
+{
+    /// <summary>
+    ///     Responsible for exporting our embedded rules so that CPS can pick them.
+    /// </summary>
+    internal static class RuleExporter
+    {
+        /// <summary>
+        ///     Represents the evaluation properties that is used for AppDesigner folder services.
+        /// </summary>
+        [ExportRule(nameof(AppDesigner), PropertyPageContexts.ProjectSubscriptionService)]
+        [AppliesTo(ProjectCapability.AppDesigner)]
```

Comment:
What's the implication of `AppliesTo` here, compared to what we had before? CPS won't make these rules available unless the capability expression matches the project's capabilities, I assume. Is there a perf buff, or is this more about correctness?

Reply:
Performance; our assembly & satellite assemblies will be loaded in scenarios it shouldn't.